models/resnet/resnet.hpp

```diff
@@ -0,0 +1,310 @@
+    inputHeight = ConvOutSize(inputHeight, kernelHeight, strideHeight, padH);
+
+    std::cout<<"Convolution: "<<inSize<<" "<<outSize<<" "<<kernelWidth<<" "<<
+        kernelHeight<<" "<<strideWidth<<" "<<strideHeight<<" "<<padW<<" "<<
+        padH<<" "<<inputWidth<<" "<<inputHeight<<std::endl;
+  }
+
+  template<typename SequentialType = ann::Sequential<>>
+  void ConvolutionBlock1x1(SequentialType* baseLayer,
+                           const size_t inSize,
+                           const size_t outSize,
+                           const size_t downSampleInputWidth = 0,
+                           const size_t downSampleInputHeight = 0,
+                           const size_t strideWidth = 1,
+                           const size_t strideHeight = 1,
+                           const size_t kernelWidth = 1,
+                           const size_t kernelHeight = 1,
+                           const size_t padW = 0,
+                           const size_t padH = 0,
+                           const bool downSample = false)
+  {
+    if (downSample)
+    {
+      baseLayer->Add(new ann::Convolution<>(inSize, outSize, kernelWidth,
+          kernelHeight, strideWidth, strideHeight, padW, padH,
+          downSampleInputWidth, downSampleInputHeight));
+
+      std::cout<<"Convolution: "<<inSize<<" "<<outSize<<" "<<kernelWidth<<" "<<
+          kernelHeight<<" "<<strideWidth<<" "<<strideHeight<<" "<<padW<<" "<<
+         padH<<" "<<downSampleInputWidth<<" "<<downSampleInputHeight<<std::endl;
+    }
+    else
+    {
+      baseLayer->Add(new ann::Convolution<>(inSize, outSize, kernelWidth,
+          kernelHeight, strideWidth, strideHeight, padW, padH, inputWidth,
+          inputHeight));
+
+      // Updating input dimesntions.
+      inputWidth = ConvOutSize(inputWidth, kernelWidth, strideWidth, padW);
+      inputHeight = ConvOutSize(inputHeight, kernelHeight, strideHeight, padH);
+      
+      std::cout<<"Convolution: "<<inSize<<" "<<outSize<<" "<<kernelWidth<<" "<<
+          kernelHeight<<" "<<strideWidth<<" "<<strideHeight<<" "<<padW<<" "<<
+          padH<<" "<<inputWidth<<" "<<inputHeight<<std::endl;
+    }
+  }
+
+  template <typename AddmergeType = ann::AddMerge<>>
+  void DownSample(AddmergeType* downSample,
+                  const size_t inSize,
+                  const size_t outSize,
+                  const size_t downSampleInputWidth,
+                  const size_t downSampleInputHeight,
+                  const size_t kernelWidth = 1,
+                  const size_t kernelHeight = 1,
+                  const size_t strideWidth = 2,
+                  const size_t strideHeight = 2,
+                  const size_t padW = 0,
+                  const size_t padH = 0)
+  {
+    ConvolutionBlock1x1(downSample, inSize, outSize, downSampleInputWidth,
+        downSampleInputHeight, strideWidth, strideHeight, kernelWidth,
+        kernelHeight, padW, padH, true);
+
+    downSample->Add(new ann::BatchNorm<>(outSize));
```

Comment:
Should batchnorm be added as third connection or as part of downsample. Wrap it as sequential and insert it into downsample layer.

Reply:
I just pushed it, can you verify if this is what you meant ?

Reply:
It worked !!! :rocket:

Reply:
Cool

Reply:
Thank you so much for spotting it.